### ISSUE TITLE
Prisma doesn't accept json field paths when trying to order by json field.

### ISSUE THREAD

**Issue by adam-arold** [prisma/prisma#10346]
### Bug description

I'm trying to order my data by a value of a field in a json column, but Prisma doesn't accept the path structure that `where` accepts. It seems that this feature of PostgreSQL is not supported yet.

### How to reproduce

Just write a query that tries to pass a `path` to the `orderBy` parameter (there is no such parameter).

### Expected behavior

I expect it to work as it works with json filtering. Something like:

```ts
orderBy: {
    myJsonField: { 
        path: ["foo", "bar"],
        orderDirection: "asc"
    },
}
```

### Prisma information

You can try this with any json field.

### Environment & setup

- OS: Windows
- Database: PostgreSQL
- Node.js version: v14.18.1

### Prisma Version

```
3.5.0
```

**Comment by homerjam** [comment]
This would be an incredibly useful feature, found this which includes some useful info for possible implementation or workaround: https://jianjye.com/articles/32/how-to-sort-mysql-json-data-correctly

**Comment by adam-arold** [comment]
> This would be an incredibly useful feature, found this which includes some useful info for possible implementation or workaround: https://jianjye.com/articles/32/how-to-sort-mysql-json-data-correctly

This is for MySQL though. What I'm thinking about is to add a regular SQL query builder and feed its result to the raw sql query option in Prisma while I'm waiting for this.

**Comment by adam-arold** [comment]
@janpio do you have a ballpark ETA for this feature?

**Comment by janpio** [comment]
No. Right now this is not a priority, which does not mean it might not be picked up by a team soon and solved. But no time to communicate.

**Comment by homerjam** [comment]
> What I'm thinking about is to add a regular SQL query builder and feed its result to the raw sql query option in Prisma while I'm waiting for this.

@adam-arold I had the same idea, Sequelize seems to be the only builder with a cast method (I am using MySQL with PlanetScale). However I'm at a very early stage with my project so would prefer to wait on this feature if it is planned to avoid the extra work of maintaining two schemas.

**Comment by sunech** [comment]
@adam-arold did you find any sort of workaround for this?
I have a scenario where I need to sort data by a postgres json field as well.

**Comment by adam-arold** [comment]
> @adam-arold did you find any sort of workaround for this? I have a scenario where I need to sort data by a postgres json field as well.

Yep. Moved away from Prisma. 😂

**Comment by sunech** [comment]
Not the answer I was hoping for 🙈 but appreciate it anyway!
+1 for this request

**Comment by adam-arold** [comment]
> Not the answer I was hoping for 🙈 but appreciate it anyway! +1 for this request

Yeah, I was upset too, but I've come to the realization that most drivers are good enough that I can use them as-is or create adapters around them.

**Comment by shtse8** [comment]
+1 for this request

**Comment by yuriipalii** [comment]
+1

**Comment by Doubl3** [comment]
Any news about this missing feature ?

**Comment by georgezouq** [comment]
+1, really need this

**Comment by cesarve78** [comment]
It seems to me a must to have feature

**Comment by adammetal** [comment]
+1, my project also needs this

**Comment by PederFosse** [comment]
+1, need this

**Comment by heozeop** [comment]
+1, want this

**Comment by cesarve77** [comment]
+1

**Comment by facuigua** [comment]
+1, this would be really helpful

**Comment by Sim1onE** [comment]
+1

**Comment by bolchowka** [comment]
+1

**Comment by banlyslash** [comment]
+1

**Comment by Ziltosh** [comment]
+1

**Comment by Kumarion** [comment]
+1

**Comment by sasithaAet** [comment]
+1

**Comment by marvin-wtt** [comment]
I have a similar issue, but with `by` from `groupBy` instead of `orderBy`. I guess this is related, or is there some way to group by a JSON field without a raw query?

**Comment by it-moysha** [comment]
Two years later, this feature is still needed...

**Comment by EFF** [comment]
We kept prisma as migration engine, and are moving to [kysely](https://kysely.dev/) as query builder. Happened to many times that prisma was standing between me and getting work done

**Comment by milos201** [comment]
Totally agree with what's been mentioned before. There have been too many times when it just gets in the way of actually getting things done. So, I've decided to create a task to replace Prisma entirely.
When it's working, it's excellent. But for more complex projects, it just has too many restrictions.

**Comment by ElYaiko** [comment]
Any update...?

**Comment by ODreelist** [comment]
It's understandably confusing as to why there's filter support but not sort support for even just top level json objects, it just seems feature incomplete. Are there any plans or progress being made in this effort? or at least a justification beyond "it's not a priority"? Thanks.

**Comment by cesarve77** [comment]
Can someone of the prisma team give us a comment. just to checking that some read our comments.

**Comment by janpio** [comment]
We read all comments. There is currently no plan to work on this in the immediate future, we have many other issues and are working on these. We are doing our best to balance that, but in this case it means this issue is currently not getting the attention some of you would prefer, sorry. We will update here when this changes.

**Comment by cesarve77** [comment]
Thanks!



[image: Mailtrack]
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
Sender
notified by
Mailtrack
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
14/06/24,
06:15:19 p.m.

**Comment by rworkduo** [comment]
I am very shocked this is still not implemented almost 3 years since it is raised

**Comment by thaua** [comment]
Has anyone got any workaround over this?

**Comment by asgeirf** [comment]
> Has anyone got any workaround over this?

A workaround... It works with TypedSQL.
https://www.prisma.io/docs/orm/prisma-client/using-raw-sql/typedsql


```prisma
model Post {
  ...
  data Json // e.g. {"category": "fishing", "priority": 1, "likes": 23}
}
```

```sql sample.sql
-- /prisma/sql/getPostsByAuthorOrderedBy.sql
-- @param {String} $1:authorId The id of the author
-- @param {String} $2:orderByField The name of the field to sort by
SELECT p.* FROM "Posts" p
  WHERE p."authorId" = $1
  ORDER BY p."data"->$2 DESC
```

```typescript
const postsByLikes = await db.$queryRawTyped(getPostsByAuthorOrderedBy(authorId, "likes"))
```

**Comment by spencer-robertson** [comment]
Any updates on this feature?
